canonical-web-and-design/maas-ui
# refactor(ui): refactor machine tests table to work with controllers

## Done

- Refactored `MachineTestsTable` and its subcomponents to work with controllers

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the React commissioning tab for a controller and see that the table data matches the AngularJS version
- Check that you can view previous tests (details page will be implemented in a follow-up)
- Go to the commissioning and test tabs for a machine and check that they work as before

## Fixes

Fixes canonical-web-and-design/app-tribe#942

## Screenshot

![Screenshot 2022-06-07 at 15-54-40 bolla commissioning bolla MAAS](https://user-images.githubusercontent.com/25733845/172306048-d7965a8a-05f0-49bc-a26a-5ab732e69e63.png)
